### PR TITLE
Feature branch

### PR DESCRIPTION
**Pull request by MarvellousUbani**

### CODE REVIEW

File: re-former/config/routes.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Rails.application.routes.draw do
+  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  resources :users, :only => [:new, :create, :edit]
```
Comment (samgaco) [inline re-former/config/routes.rb]: - [ ] we should have a route for `update` as well

File: re-former/app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+class UsersController < ApplicationController
+
+	def new
+		@user = User.new
+	end
+
+	def create
+		  # @user = User.new(username: params[:username], email: params[:email], password: params[:password])
```
Comment (samgaco) [inline re-former/app/controllers/users_controller.rb]: - [ ] try to remove commented code in your app

File: re-former/app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      if @user.save
+        redirect_to new_user_path
+      else
+        render :new
+      end
+	end
+
+	def edit
+		@user = User.find(params[:id])
+	end
```
Comment (samgaco) [inline re-former/app/controllers/users_controller.rb:18]: - [ ] we should have an update method as well, so we can update the information of our user when we edit.

File: re-former/app/views/users/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,32 @@
+<!-- <form action="/users" method="post" accept-charset="UTF-8" >
+  <label for="username"> Type in username</label> <br>
```
Comment (samgaco) [inline re-former/app/views/users/new.html.erb:2]: - [ ] you should have `form_tag` commented as well

> Railsy Forms with #form_tag
> Now we’ll start morphing our form into a full Rails form using the #form_tag and #*_tag helpers. There’s actually very little additional help that’s going on and you’ll find that you’re mostly just renaming HTML tags into Rails tags.

**Review by samgaco — CHANGES_REQUESTED** [review]
Hi @MarvellousUbani and @Ashah15 :high_brightness: 

Right now your edit functionality is still nor working properly if we try to update the information of a user we have created we are getting some routing errors. Please, before working on this try to assess all the comments I left below and then ensure "edit" and "update" are well implemented so your users are able to edit their data and actualize it in the server database.

Good spirit! :muscle: 

-------------------------------
Moving on :arrow_down: 

* **[For your next code review requests]**  Add a short summary of changes in Pull Request description
* **[For your next code review requests]**  Use meaningful names for your Pull Requests

-----------------------

* ###### STATUS OF THE PROJECT: REQUESTED CHANGES :stop_sign:

**Comment by MarvellousUbani** [comment]
> Hello and Thanks for your feedback.
We've been able to add in the update method on the users controller and also test that it works well. 
We've also removed the specified comment from the code and added in the form_tag method commented out as well.

File: re-former/app/models/user.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class User < ApplicationRecord
+	validates :username, presence:true
```
Comment (mosaaleb) [inline re-former/app/models/user.rb:2]: - We can validate multiple attributes in the same line if they all share the same validator
`validates :username, :email, :password, presence: true`

**Review by mosaaleb — APPROVED** [review]
## Well done @MarvellousUbani, @Ashah15 :clap:


#### Good job implementing the previous review requirements. All requirements are handled :100:

#### Please Consider my comments before moving on:


- [ ] `new.html.erb` and `edit.html.erb` have the same code which is repeating. To DRY these views using 2 [partial](https://guides.rubyonrails.org/layouts_and_rendering.html#using-partials) files:

  - one file is to put all the error messages in. -- let's add it.
  - the other file is to put the `form_for` as it has the same content in both views.


#### You can merge :+1:

#### Good luck with the next projects.

##### Muhammad Ebeid

**Pull request merged**